site/src/sphinx/advanced-retry.rst

```diff
@@ -1,8 +1,228 @@
+- Random delay, created with ``Backoff.random()``
+- Exponential delay which is multiplied on each attempt, created with ``Backoff.exponential()``
+
+Armeria provides ``RetryStrategy.defaultBackoff`` that you might use as default. It is exactly same with:
+
+.. code-block:: java
+
+    Backoff.exponential(minDelayMillis, maxDelayMillis, multiplier).withJitter(jitterRate);
+
+The delay starts from ``minDelayMillis`` until it reaches ``maxDelayMillis`` multiplying with multiplier.
+Please note that the ``.withJitter()`` will add jitter value to the calculated delay.
+For more information, please refer to the API documentation of the `com.linecorp.armeria.client.retry`_ package.
+
+.. _retry-with-logging:
+
+RetryingClient with logging
+---------------------------
+
+You can use RetryingClient_ with LoggingClient_ to log. If you want to log all of the requests and responses,
+decorate LoggingClient_ with RetryingClient_. That is:
+
+.. code-block:: java
+
+    RetryStrategy strategy = RetryStrategy.onServerErrorStatus();
+    HttpClient client = new HttpClientBuilder(...)
+                              .decorator(LoggingClient.newDecorator())
+                              .decorator(RetryingHttpClient.newDecorator(strategy))
+                              .build();
+
+This will produce following logs when there's three attempts:
+
+.. code-block:: java
+
+    LoggingClient - Request: {startTime=..., length=..., duration=..., scheme=..., host=..., headers=[...]
+    LoggingClient - Response: {startTime=..., length=..., duration=..., headers=[:status=500, ...]
+    LoggingClient - Request: {startTime=..., length=..., duration=..., scheme=..., host=..., headers=[...]
+    LoggingClient - Response: {startTime=..., length=..., duration=..., headers=[:status=500, ...]
+    LoggingClient - Request: {startTime=..., length=..., duration=..., scheme=..., host=..., headers=[...]
+    LoggingClient - Response: {startTime=..., length=..., duration=..., headers=[:status=200, ...]
+
+If you want to log the first request and the last response, no matter if it's successful or not,
+do the reverse:
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.logging.LoggingClient;
+
+    RetryStrategy strategy = RetryStrategy.onServerErrorStatus();
+    HttpClient client = new HttpClientBuilder(...)
+                              .decorator(RetryingHttpClient.newDecorator(strategy))
+      /* notice the order */  .decorator(LoggingClient.newDecorator())
+                              .build();
+
+This will produce only single request and response log pair regardless how many attempts are made:
+
+.. code-block:: java
+
+    LoggingClient - Request: {startTime=..., length=..., duration=..., scheme=..., host=..., headers=[...]
+    LoggingClient - Response: {startTime=..., length=..., duration=..., headers=[:status=200, ...]
+
+This is achieved using the feature of :ref:`nested-log`. For more information, please refer to
+:ref:`nested-log`.
```

Comment:
I guess these two sentences could be simplified a little bit so that 'nested-log' is referred only once.